hypothesis-python/src/hypothesis/internal/conjecture/utils.py

```diff
@@ -60,6 +60,23 @@ def combine_labels(*labels):
 BIASED_COIN_LABEL = calc_label_from_name("biased_coin()")
 SAMPLE_IN_SAMPLER_LABLE = calc_label_from_name("a sample() in Sampler")
 ONE_FROM_MANY_LABEL = calc_label_from_name("one more from many()")
+SAMPLER_FOR_BITS_CACHE = {}
+
+
+def sampler_for_bit_sizes(max_bits=128):
+    if max_bits in SAMPLER_FOR_BITS_CACHE:
+        return SAMPLER_FOR_BITS_CACHE[max_bits]
+    # All bit sizes [8, 16, 32, .. max_bits], to be sampled from
+    sizes = [2 ** i for i in range(3, bit_length(max_bits) + 1)]
+    # Explicit weights for up to 256-bit numbers; then constant low weight
+    weights = [4.0, 8.0, 1.0, 1.0, 0.5, 0.1] + [0.01] * 30
```

Comment:
What's the significance of the 30 here? Granted we're probably never going to have 68719476736-bit numbers, but would it not make more sense to extend the list of weights to match the size rather than have some arbitrary large size and truncate it down?

Reply:
30 is just "larger than we'll ever generate". I thought it was simpler to truncate given that sizes could have as few as two or three elements.

Todo: check that it's non-empty to guard against future changes.

Reply:
Yeah I think it's unlikely to matter in practice, it's just a sort of code that makes me a bit twitchy. I'd be fine if you added an assert or something too.

Reply:
To clarify, hitting this would require Hypothesis to be using a several-hundred-gigabyte buffer to generate a single integer.  I'm not too worried about that!  (but asserts added)